File: leetcode/Easy/csharp/100. Same Tree.cs
/*
https://leetcode.com/problems/same-tree/description/

Автор: Дуплей Максим Игоревич
ORCID: https://orcid.org/0009-0007-7605-539X
GitHub: https://github.com/QuadDarv1ne/
*/

public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {
        /*
        Идея:
        - Если оба узла null — одинаковы.
        - Если один null — не одинаковы.
        - Иначе сравниваем значения и рекурсивно левое/правое.
        */
        if (p == null && q == null) return true;
        if (p == null || q == null) return false;
        if (p.val != q.val) return false;
        return IsSameTree(p.left, q.left) && IsSameTree(p.right, q.right);
    }
}

File: leetcode/Easy/cpp/100. Same Tree.cpp
/*
https://leetcode.com/problems/same-tree/description/

Автор: Дуплей Максим Игоревич
ORCID: https://orcid.org/0009-0007-7605-539X
GitHub: https://github.com/QuadDarv1ne/
*/

class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        /*
        Решение задачи "Same Tree" (LeetCode 100).

        Идея:
        - Проверяем оба узла на NULL.
        - Если не NULL, сравниваем значения и рекурсивно левые/правые.
        */
        if (!p && !q) return true;
        if (!p || !q) return false;
        if (p->val != q->val) return false;
        return isSameTree(p->left, q->left) && isSameTree(p->right, q->right);
    }
};
